src/main/java/com/ncc/aif/ValidateAIF.java

```diff
@@ -0,0 +1,311 @@
+package com.ncc.aif;
+
+import ch.qos.logback.classic.Level;
+import ch.qos.logback.classic.Logger;
+import com.google.common.base.Charsets;
+import com.google.common.collect.ImmutableList;
+import com.google.common.io.CharSource;
+import com.google.common.io.Files;
+import com.google.common.io.Resources;
+import edu.isi.nlp.parameters.Parameters;
+import org.apache.jena.query.*;
+import org.apache.jena.rdf.model.*;
+import org.apache.jena.util.FileUtils;
+import org.apache.jena.vocabulary.RDF;
+import org.topbraid.shacl.validation.ValidationUtil;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.Reader;
+import java.util.HashSet;
+
+
+public final class ValidateAIF {
+
+    private static final String SHACL_RESNAME = "com/ncc/aif/aida_ontology.shacl";
+    private static final String TA3_SHACL_RESNAME = "com/ncc/aif/aida_hypothesis.shacl";
+    private static final String INTERCHANGE_RESNAME = "com/ncc/aif/InterchangeOntology";
+    private static final String AIDA_DOMAIN_COMMON_RESNAME = "com/ncc/aif/AidaDomainOntologiesCommon";
+
+    private Model domainModel;
+    private static Model shaclModel;
+    private static Model ta3ShaclModel;
+
+    private ValidateAIF(Model domainModel) {
+        this.domainModel = domainModel;
+        try {
+            shaclModel = loadModel(Resources.asCharSource(Resources.getResource(SHACL_RESNAME), Charsets.UTF_8)
+                    .openBufferedStream());
+        } catch (IOException ioe) {
+            throw new RuntimeException("While parsing AIDA Shacl " + SHACL_RESNAME, ioe);
+        }
+
+        try {
+            ta3ShaclModel = loadModel(Resources.asCharSource(Resources.getResource(TA3_SHACL_RESNAME), Charsets.UTF_8)
+                    .openBufferedStream());
+        } catch (IOException ioe) {
+            throw new RuntimeException("While parsing TA3 Shacl " + SHACL_RESNAME, ioe);
+        }
+    }
+
+    private static Model loadModel(Reader reader) {
```

Comment:
My approach for AIDA-447 was not to change the code unless there was a Java reason to do so, it pertained to ColdStart, or it otherwise caused problems.  I suggest we defer this kind of change to AIDA-500.

Reply:
In general I usually forget to make these types of changes if I don't make them when I notice them. As you already have a task for these types of things, I'm happy to defer

Reply:
I added a note to AIDA-500.